packages/repository/src/__tests__/unit/model/model.unit.ts

```diff
@@ -394,6 +394,56 @@ describe('model', () => {
     });
   });
 
+  it('converts to plain object including relation properties', () => {
+    class Category extends Entity {
+      id: number;
+      products: Product[];
+
+      // allow additional (free-form) properties
+      // eslint-disable-next-line @typescript-eslint/no-explicit-any
+      [propName: string]: any;
+
+      constructor(data: Partial<Category>) {
+        super(data);
+      }
+    }
+
+    class Product extends Entity {
+      id: number;
+      categoryId: number;
+
+      constructor(data: Partial<Product>) {
+        super(data);
+      }
+    }
+
+    Category.definition = new ModelDefinition('Category')
+      .addSetting('strict', false)
+      .addProperty('id', {type: 'number', id: true, required: true})
+      .addRelation(<HasManyDefinition>{
+        name: 'products',
+        type: RelationType.hasMany,
+        targetsMany: true,
+
+        source: Category,
+        keyFrom: 'id',
+
+        target: () => Product,
+        keyTo: 'categoryId',
+      });
+
+    const category = new Category({
```

Comment:
Theoretically, we don't allow to create an instance alone with navigational props (`products`). But I think it's fine here as we aim to test `toObject`.

Reply:
Thanks @agnes512 good reminder, the test won't create an instance with navigational property, it mocks the composed result, like a query with inclusion filter.